server/config.go

```diff
@@ -163,13 +166,38 @@ func (cfg *SingleServerConfig) Configure(srv *Server) error {
 	return nil
 }
 
-func loadUsers(filepath string) (users []user.UserWithRemoteIdentities, err error) {
+// loadUsers parses the user.json file and returns the users to be created.
+func loadUsers(filepath string) (users []user.UserWithRemoteIdentities, pwis []user.PasswordInfo, err error) {
```

Comment:
Stuff like this is much easier to test if you separate out the concept of loading from a file from reading from a reader. In other words, there should be a loadUsersFromReader method which does most of the heavy lifting here, and then you dont have to do this tempfile stuff in tests. See for example NewEmailerConfigFromReader